src/dialog-controller.js

```diff
@@ -58,7 +58,9 @@ export class DialogController {
    * @returns Promise An empty promise object.
    */
   close(ok: boolean, output?: any): Promise<DialogResult> {
-    if (this._closePromise) return this._closePromise;
+    if (this._closePromise) { 
+      return this._closePromise;
+    }
 
     this._closePromise = invokeLifecycle(this.viewModel, 'canDeactivate').then(canDeactivate => {
```

Comment:
@PWKad When I wrote it I wasn't sure if unresolved promises cause memory leaks and that was my reasoning for wanting to call `.error()` - checked, don't cause.
Also `.error` will call `invokeLifecycle(this.viewModel, 'deactivate')`. Is it OK for `deactivate` to be called when `canDeactivate` has failed? If it's OK, I'll test it with calling `.error`.